### PR TITLE
[DOCFIX] Update S3 API documentation

### PR DESCRIPTION
**Pull request by ZhuTopher**
### What changes are proposed in this pull request?

Updates the [S3 API doc](https://docs.alluxio.io/os/user/stable/en/api/S3-API.html) to reflect recent changes.

### Why are the changes needed?

New features & behaviour has been added to Alluxio's S3-compatible REST API in #15205 and #15333
This DOCFIX updates the docs to reflect that.

### Does this PR introduce any user facing changes?

Updated contents for the Alluxio [S3 API doc](https://docs.alluxio.io/os/user/stable/en/api/S3-API.html).

### CODE REVIEW

File: docs/en/api/S3-API.md [hunk 1]
```diff
@@ -23,61 +23,113 @@ Alluxio proxy, introducing an extra hop. For optimal performance, it is recommen
...
 
-Alluxio s3 will overwrite the existing key and the temporary directory for multipart upload. 
+Alluxio S3 will overwrite the existing key and the temporary directory for multipart upload.
 
 
-## Features support
-The following table describes the support status for current Amazon S3 functional features:
+## Supported S3 API Actions
+
+Global request headers: **Insert table here -> Authorization**
```
Comment (Xenorith) [inline docs/en/api/S3-API.md:35]: placeholder for next update?

File: docs/en/api/S3-API.md [hunk 2]
```diff
@@ -23,61 +23,113 @@ Alluxio proxy, introducing an extra hop. For optimal performance, it is recommen
...
 For example, you can run the following RESTful API calls to an Alluxio cluster running on localhost.
 The Alluxio proxy is listening at port 39999 by default.
 
 #### Authorization
 
-By default, the user that is used to do any FileSystem operations is the user that was used to
-launch the proxy process. This can be changed by providing the Authorization Header. The header
-format is [defined by the AWS S3 Rest API reference](https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-auth-using-authorization-header.html)
+At the moment, access key and secret key validation does not exist for the Alluxio S3 API.
+Therefore the 'Authorization' header is used purely to specify the intended user to perform a request.
```
Comment (Xenorith) [inline docs/en/api/S3-API.md:110]: single quote intentional or should be backtick?

File: docs/en/api/S3-API.md (hunk 1 above)
Comment (ZhuTopher) [reply docs/en/api/S3-API.md]: Was a placeholder for _this_ PR

File: docs/en/api/S3-API.md (hunk 2 above)
Comment (ZhuTopher) [reply docs/en/api/S3-API.md]: Outdated

File: docs/en/api/S3-API.md [hunk 3]
```diff
@@ -18,375 +18,1701 @@ sharing with file system based applications, and storage system abstraction (e.g
...
+      {% for param in params %}
+      {% if forloop.last %}
+      <li>{{ param }}</li>
+      {% else %}
+      <li>{{ param }},</li>
+      {% endif %}
+      {% endfor %}
+      </ul>
+      {% endif %}
+    </td>
```
Comment (ZhuTopher) [inline docs/en/api/S3-API.md:142]: The for loops in these two `<td>` blocks are just trying to process the `|`-separated entries of the csv.

The if-else blocks are just handling whether or not to append a trailing `,`.

File: docs/en/api/S3-API.md [hunk 4]
```diff
@@ -18,375 +18,1701 @@ sharing with file system based applications, and storage system abstraction (e.g
...
+  </tr>
+  {% endif %}
+{% endfor %}
+</table>
 
 ## Example Usage
 
-### REST API
-For example, you can run the following RESTful API calls to an Alluxio cluster running on localhost.
-The Alluxio proxy is listening at port 39999 by default.
```
Comment (ZhuTopher) [inline docs/en/api/S3-API.md:62]: The entire `REST API` examples section was collapsed into `navtabs` alongside `aws` cli examples.

They were mapped to be one-to-one with their corresponding S3 API Actions.

File: docs/en/api/S3-API.md [hunk 5]
```diff
@@ -18,375 +18,1701 @@ sharing with file system based applications, and storage system abstraction (e.g
...
+    <LastModified>2022-05-03T16:48:56.602Z</LastModified>
+    <ETag>""</ETag>
+    <Size>27040</Size>
+  </Part>
+</ListPartsResult>
 ```
+{% endnavtab %}
+{% endnavtabs %}
 
 ### Python S3 Client
```
Comment (ZhuTopher) [inline docs/en/api/S3-API.md:1717]: Python S3 client examples are untouched and should be considered out-of-date (eg: they are missing examples for tagging)

File: docs/_data/table/s3-api-supported-actions.csv [hunk 6]
```diff
@@ -0,0 +1,24 @@
+action,endpoint,headers,queryParams
+AbortMultipartUpload,POST /{bucket}/{object}?uploadId={UploadId},None,
+CompleteMultipartUpload,POST /{bucket}/{object}?uploadId={UploadId},None,
+CopyObject,PUT /{bucket}/{object},Content-Type|x-amz-copy-source|x-amz-metadata-directive|x-amz-tagging-directive|x-amz-tagging,
```
Comment (ZhuTopher) [inline docs/_data/table/s3-api-supported-actions.csv:4]: I used `|` as a list-separator. Not sure if there is a preferred choice.

File: docs/en/api/S3-API.md [hunk 7]
```diff
@@ -18,375 +18,1701 @@ sharing with file system based applications, and storage system abstraction (e.g
...
+The following table describes the support status for current [S3 API Actions](https://docs.aws.amazon.com/AmazonS3/latest/API/API_Operations.html):
 
 <table class="table table-striped">
-  <tr><th>S3 Feature</th><th>Status</th></tr>
-{% for item in site.data.table.s3-api-supported-operations %}
   <tr>
-    <td>{{ item.S3Feature }}</td>
-    <td>{{ item.Status }}</td>
+    <th>S3 API Action</th>
+    <th>REST API Endpoint</th>
```
Comment (ZhuTopher) [inline docs/en/api/S3-API.md:103]: @Xenorith In hindsight, I think including this alone is insufficient as some endpoints are shared (with the same query parameters) but use different request bodies.

I think I should remove this column entirely (as they should look at the S3 documentation for details). wdyt?

Comment (ZhuTopher) [reply docs/en/api/S3-API.md]: Removed

File: docs/en/api/S3-API.md [hunk 8]
```diff
@@ -18,375 +18,1624 @@ sharing with file system based applications, and storage system abstraction (e.g
...
+You can use the [AWS command line interface](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-welcome.html)
+to send S3 API requests to the Alluxio S3 API. Note that you will have to provide the `--endpoint` parameter
+to specify the location of the Alluxio S3 REST API with the server's base URI included
+(i.e: `--endpoint "http://{alluxio.proxy.web.hostname}:{alluxio.proxy.web.port}/api/v1/s3/"`).
 
-By default, the user that is used to do any FileSystem operations is the user that was used to
-launch the proxy process. This can be changed by providing the Authorization Header. The header
-format is [defined by the AWS S3 Rest API reference](https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-auth-using-authorization-header.html)
+As a pre-requisite for operations which involve the `Authorization` header you may need to [configure
+some AWS credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html).
```
Comment (Xenorith) [inline docs/en/api/S3-API.md:176]:
```suggestion
AWS credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html).
```

File: docs/en/api/S3-API.md (hunk 5 above)
Comment (Xenorith) [reply docs/en/api/S3-API.md:1640]: i assume the existing python example still works as described; the incoming changes should have no effect on this?

**Review by Xenorith — APPROVED** [review]
structure looks much better, a few nits otherwise good

File: docs/en/api/S3-API.md [hunk 9]
```diff
@@ -18,375 +18,1699 @@ sharing with file system based applications, and storage system abstraction (e.g
...
+### Tagging & Metadata Limits
+
+User-defined tags on buckets & objects are limited to 10 and obey the [S3 tag restrictions](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-tagging.html).
+
+The maximum size for user-defined metadata in PUT-requests is 2KB by default in accordance with [S3 object metadata restrictions](https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html).
+See the property key `alluxio.proxy.s3.header.metadata.max.size` to change this behavior.
+
+### Performance Implications
+
+The S3 API leverages the [Alluxio REST proxy]({{ '/en/api/FS-API.html#rest-api' | relativize_url }}), introducing an extra hop.
```
[the remark this answers is not in the document]
Comment (ZhuTopher) [reply docs/en/api/S3-API.md]: Changed to "an additional network hop for Alluxio clients."

Network hop is the term both colloquially and technically AFAIK.

**Comment by Xenorith** [comment]
alluxio-bot, merge this please

**Comment by Xenorith** [comment]
alluxio-bot, cherry-pick this to branch-2.8 please

**Pull request merged**